### PR TITLE
map feature branches to master tag for verify

### PR DESCRIPTION
**Pull request by kow3ns**
reduce special handling to handling for release branches.

### CODE REVIEW

File: scenarios/kubernetes_verify.py [hunk 1]
```diff
@@ -52,16 +52,25 @@ def check(*cmd):
     print >>sys.stderr, 'Run:', cmd
     subprocess.check_call(cmd)
 
+def get_tag(branch):
+    """Gets the tag associated with a branch."""
+    if branch == 'master':
+        return BRANCH_VERSION.get('master')
```
Comment (enisoc) [inline scenarios/kubernetes_verify.py]: I think this should be:

```go
return VERSION_TAG[BRANCH_VERSION.get('master')]
```

File: scenarios/kubernetes_verify.py [hunk 2]
```diff
@@ -52,16 +52,25 @@ def check(*cmd):
...
+
+    release = re.match(r'release-(\d+\.\d+)', branch)
```
Comment (enisoc) [inline scenarios/kubernetes_verify.py]: Please copy over the comment about 3-part versions for future reference. I added it to explain the fix in #2801.

File: scenarios/kubernetes_verify.py [hunk 3]
```diff
@@ -52,16 +52,25 @@ def check(*cmd):
...
+    if release:
+        ver = release.group(1)
+        return VERSION_TAG[BRANCH_VERSION.get(ver, ver)]
+
+    feature = re.match(r'feature-\d+', branch)
+    if feature:
+        return BRANCH_VERSION.get('master')
```
Comment (enisoc) [inline scenarios/kubernetes_verify.py]: Same as above.

**Comment by enisoc** [comment]
Could we just say any unrecognized branch is treated the same as master?

**Comment by fejta** [comment]
Works for me. Sen has been doing the most work in this area

**Comment by krzyzacy** [comment]
we are deprecating this scenario and the node image - all logic should move to [kubetest](https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L509)

**Comment by krzyzacy** [comment]
(I'll close this - we are migrate the node jobs off Jenkins and will delete this file very soon)

**Comment by kow3ns** [comment]
@krzyzacy do you have a time frame?

**Comment by krzyzacy** [comment]
@kow3ns sorry did not see the verify part - I would say next Mon for kubelet, and Q4 for verify.

@BenTheElder we can get this in, since I'm not sure when we can move the verify job to prow yet?

File: scenarios/kubernetes_kubelet.py [hunk 4]
```diff
@@ -53,14 +53,20 @@ def var(path):
     """Expands '${foo} interesting' to 'something interesting'."""
     return os.path.expandvars(path)
 
+def get_tag(branch):
+    #If branch 3-part version, only take the first two parts
+    release = re.match(r'release-(\d+\.\d+)', branch)
```
Comment (enisoc) [inline scenarios/kubernetes_kubelet.py]: For some reason, in this file the `VERSION_TAG` keys include the `release-` prefix. Not sure if anything else is depending on that...

Comment (krzyzacy) [reply scenarios/kubernetes_kubelet.py]: hummm, good catch, yeah the image tag is different

Comment (krzyzacy) [reply scenarios/kubernetes_kubelet.py]: @kow3ns to match previous behavior I think you want to use `match.group(0)` here instead

Comment (kow3ns) [reply scenarios/kubernetes_kubelet.py]: @krzyzacy 
Match group(1) contains the version (e.g. 1.2 or 1.3). BRANCH_VERSION gets version associated with the branch and VERSION_TAG gets the tag associated with that version.

Comment (krzyzacy) [reply scenarios/kubernetes_kubelet.py]: okay, then you'll need to fix `VERSION_TAG` above as well (trim `release-` part)

Comment (krzyzacy) [reply scenarios/kubernetes_kubelet.py]: I think for `release-1.7`, `VERSION_TAG[BRANCH_VERSION.get('1.7', '1.7')]` will not have an entry now?

**Pull request closed without merging**